### DISCUSSION TITLE
Problem about compiling and testing application

### DISCUSSION THREAD

**Discussion by AimZhang** [idaholab/moose#22026] [Q&A Getting Started] [answered]
Hello, i am the new beginner to use MOOSE, when I compile and test the application, I met the issue (one error: no matching function for call to 'index_range' for (const auto pid : index_range(mesh->comm()))). Please see the attached. Could you please give me some assistance on that? Thank you very much.

<img width="1512" alt="errors" src="https://user-images.githubusercontent.com/110468396/188750183-4ad78ff2-eab8-4385-8b4f-4001afab0bd3.png">

**Comment by GiudGiud** [comment, marked as answer]
Hello

this is an incompatibility between your moose and libmesh version. 

I recommend you update both, following the procedure on the website

Guillaume

**Reply by AimZhang** [reply]
Thank you very much for your comments. I know how to update MOOSE according to the procedure of the website. But I am not sure how to update libmesh, I just downloaded the latest version of libmesh. And then I tried to compile and test the application, i got the same error. If I was wrong with updating libmesh, could you please show the detailed information about updating libmesh? Thank you very much.

**Reply by GiudGiud** [reply]
see this page
https://mooseframework.inl.gov/getting_started/installation/update_moose.html
the conda update will update libmesh

**Reply by AimZhang** [reply]
Thank you very much for your suggestions and guidance. Now the problem has been figured out. I have succeeded in testing the application.
![succeed](https://user-images.githubusercontent.com/110468396/188891980-801da351-711d-4aba-94e6-7d622dbafcc7.png)
